File: boosting/src/data/WeightedData.java
package data;

public class WeightedData implements RegressionData {

    private double weight;
    private final RegressionData data;

    public WeightedData(RegressionData data) {
        this(data, 1);
    }

    public WeightedData(RegressionData data, double weight) {
        this.data = data;
        this.weight = weight;
    }

    public void setWeight(double weight) {
        this.weight = weight;
    }

    @Override
    public double[] asVector() {
        return data.asVector();
    }

    @Override
    public double output() {
        return data.output() * weight;
    }
}

File: boosting/src/data/WeightedData.kt
package data

class WeightedData @JvmOverloads constructor(private val data: RegressionData, private var weight: Double = 1.0) : RegressionData {

    fun setWeight(weight: Double) {
        this.weight = weight
    }

    override fun asVector(): DoubleArray {
        return data.asVector()
    }

    override fun output(): Double {
        return data.output() * weight
    }
}
